crates/fluvio-connector-package/src/secret.rs

```diff
@@ -75,6 +90,34 @@ impl From<Secret> for SecretString {
 
 pub trait SecretStore: Send + Sync {
     fn read(&self, name: &str) -> Result<String>;
+
+    fn resolve_str(&self, input: &str) -> Result<String> {
+        let mut content = String::new();
+        let mut last = 0;
+
+        for cap in get_or_init_secrets_re()?.captures_iter(input) {
+            let range = cap.get(0).ok_or(anyhow!(
+                "Unreachable error, capture should have at least a match"
+            ))?;
+            content.push_str(&input[last..range.start()]);
+            let secret_name = cap
+                .get(1)
+                .ok_or(anyhow!(
+                    "Unreachable error, capture has defined a non-empty capture group"
```

Comment:
same.  need to hide regex related which is only relevant to regex implementation